src/Settings.php

```diff
@@ -350,6 +380,11 @@ public static function restoreDefaultsAction()
     
     public static function flashRedirect($type, $message)
     {
+        // Security check - ensure user has proper capabilities
+        if (!current_user_can('manage_options')) {
```

Comment:
Is there a reason we are adding this here when it is only called from `restoreDefaultsAction()` which has the same check?